core/shared/src/main/scala/org/typelevel/log4cats/PagingSelfAwareStructuredLogger.scala

```diff
@@ -52,12 +53,26 @@ object PagingSelfAwareStructuredLogger {
    * @return
    *   SelfAwareStructuredLogger with pagination.
    */
-  def withPaging[F[_]: Monad](pageSizeK: Int = 64, maxPageNeeded: Int = 999)(
+  def withPaging2[F[_]: Monad: UUIDGen](pageSizeK: Int = 64, maxPageNeeded: Int = 999)(
       logger: SelfAwareStructuredLogger[F]
   ): SelfAwareStructuredLogger[F] =
-    new PagingSelfAwareStructuredLogger[F](pageSizeK, maxPageNeeded)(logger)
+    new PagingSelfAwareStructuredLogger[F](pageSizeK, maxPageNeeded, UUIDGen.randomUUID)(logger)
 
-  private class PagingSelfAwareStructuredLogger[F[_]: Monad](pageSizeK: Int, maxPageNeeded: Int)(
+  @deprecated("Use withPaging2", "2.5.0")
+  def withPaging[F[_]: Monad](pageSizeK: Int = 64, maxPageNeeded: Int = 999)(
+      logger: SelfAwareStructuredLogger[F]
+  ): SelfAwareStructuredLogger[F] =
+    new PagingSelfAwareStructuredLogger[F](
+      pageSizeK,
+      maxPageNeeded,
+      Monad[F].unit.map(_ => UUID.randomUUID())
```

Comment:
Terrible, right! But this just more clearly exposes the bad behavior that already existed here. There is no safe way to generate random `UUID`s with only a `Monad`.

Reply:
Absolutely! I'm just sorry I didn't catch it before